imitation/util/buffer.py

```diff
@@ -0,0 +1,213 @@
+from typing import Optional, Tuple
```

Comment:
Overall this is a very clean implementation :) It feels like both `Buffer` and `ReplayBuffer` should be instances of some abstract class. Consider renaming `Buffer` to e.g. `FIFOBuffer` and making a new `Buffer` abstract class?

Reply:
I felt like that too, but most of the methods in `Buffer` have different arguments. `abc.ABC` will force you define this methods, but it feels icky to me that I would be overriding the `ABC` arguments, or setting the `ABC` arguments to something like `f(*args, **kwargs)`.

Willing to revisit this sort of refactoring later, but I don't think this is a priority now.

Reply:
Let me know if there is an good ABC style that would cover both `store(self, data: np.ndarray) -> None:` and `store(self, old_obs: np.ndarray, act: np.ndarray, new_obs: np.ndarray):`.

Reply:
I agree the different types make it awkward. I think `store(self, *args: List[np.array])` isn't crazy -- you're in both cases storing one or more NumPy arrays. But I'm willing to leave this for now.